libr/bin/demangle.c

```diff
@@ -295,21 +295,87 @@ R_API char *r_bin_demangle_objc(RBinFile *binfile, const char *sym) {
 	return ret;
 }
 
+static bool replace_seq (const char **in, char **out, const char *seq, char value) {
+	size_t len = strlen (seq);
+
+	if (strncmp (*in, seq, len)) {
+		return false;
+	}
+
+	**out = value;
+
+	*in += len;
+	*out += 1;
+
+	return true;
+}
+
+R_API char *r_bin_demangle_rust (RBinFile *binfile, const char *sym, ut64 vaddr) {
+	size_t len;
+	char *str, *out, *in;
+
+	str = r_bin_demangle_cxx (binfile, sym, vaddr);
+	out = in = str;
+	len = strlen (str);
+
+	if (*in == '_') {
+		*in++;
+	}
+
+	while (len-- > 0) {
+		if (!(*in == '$' && (replace_seq (&in, &out, "$SP$", '@')
```

Comment:
this can be simplified by using a macro

#define RS(x,y) replace_seq(&in,&out,x,y)